src/Nfts/Nfts.tsx

```diff
@@ -4,23 +4,34 @@ import {RefreshControl, ScrollView, StyleSheet, Text, View} from 'react-native'
 import {SafeAreaView} from 'react-native-safe-area-context'
 
 import {Icon, Spacer} from '../components'
-import {useFilteredNfts} from './hooks'
+import {useSearch} from '../Search/SearchContext'
+import {useSelectedWallet} from '../SelectedWallet'
+import {useNfts, YoroiNft} from '../yoroi-wallets'
+import {filterNfts} from './filterNfts'
 import {ImageGallery, SkeletonGallery} from './ImageGallery'
 import {useNavigateTo} from './navigation'
 import {NoNftsScreen} from './NoNftsScreen'
 
 export const Nfts = () => {
   const navigateTo = useNavigateTo()
-  const handleNftSelect = (index: number) => navigateTo.nftDetails(nfts[index].id)
   const [isManualRefreshing, setIsManualRefreshing] = React.useState(false)
+  const wallet = useSelectedWallet()
   const strings = useStrings()
 
-  const {search, nfts, isLoading, refetch, isError} = useFilteredNfts({
+  const {isLoading, nfts, refetch, isError} = useNfts(wallet, {
     onSettled: () => {
       if (isManualRefreshing) setIsManualRefreshing(false)
     },
   })
 
+  const {search: nftsSearchTerm} = useSearch()
+  const sortedNfts = sortNfts(nfts)
+  const nftsSearchResult = filterNfts(nftsSearchTerm, sortedNfts)
+
+  const handleNftSelect = (index: number) => navigateTo.nftDetails(nftsSearchResult[index].id)
```

Comment:
don't use index

Reply:
yap, I think it can receive the `id` directly.

Reply:
what name would you suggest? it is the index of the nfts array

Reply:
not the name, i mean literally, dont use indexes

Reply:
`onSelect(index: number)` -> `onSelect(id: string)`

Reply:
the selection process is `nft -> index -> nft -> id`
just pass the `id`